server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Properties.scala

```diff
@@ -50,6 +51,7 @@ case class Properties(value: Set[NonEmptyString]) {
   def filter(o: JsObject): JsObject =
     JsObject(o.fields.filter(entry => {
       val refined: Either[String, NonEmptyString] = refineV[NonEmpty](entry._1)
-      refined.fold(e => throw new RuntimeException(e), property => contains(property))
+      refined.fold(e => throw new RuntimeException(e),
+        property => value.map(property => sanitizeHeader(property)).contains(property.value))
```

Comment:
Again, ` sanitizeHeader` call have nothing to do here, in a generic model.

(That's the beginning of a spaggetthi)